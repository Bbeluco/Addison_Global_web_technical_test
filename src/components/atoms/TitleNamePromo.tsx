import React from 'react';
import {Text, View} from 'react-native';
import {DescribePromoType} from '../../types/describePromoType';

export function TitleNamePromo({
  text,
  styleChildren,
}: DescribePromoType): React.JSX.Element {
  return (
    <View>
      <Text style={styleChildren}>{text}</Text>
    </View>
  );
}
