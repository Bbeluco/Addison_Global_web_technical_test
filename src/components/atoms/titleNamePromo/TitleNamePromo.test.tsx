import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {TitleNamePromo} from './TitleNamePromo';

describe('<TitleNamePromo />', () => {
  it('should render and reflect the text passed as props', () => {
    const text = 'Unit test';
    render(<TitleNamePromo styleChildren={{}} text={text} />);
    expect(screen.queryByText(text)).not.toBeNull();
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
