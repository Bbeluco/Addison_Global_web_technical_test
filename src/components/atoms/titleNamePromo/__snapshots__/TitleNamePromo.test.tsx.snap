// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleNamePromo /> should render and reflect the text passed as props 1`] = `
<View>
  <Text
    style={{}}
  >
    Unit test
  </Text>
</View>
`;
